src/Verify.MSTest.SourceGenerator/ClassToGenerate.cs

```diff
@@ -0,0 +1,13 @@
+readonly record struct ClassToGenerate
+{
+    public string? Namespace { get; }
+    public string ClassName { get; }
+    public EquatableArray<ParentClass> ParentClasses { get; }
```

Comment:
cant we rely on the gethashcode and equality generated methods that we get via record struct? then we dont need EquatableArray

Reply:
We can't rely on the autogenerated ones, as they defer equality of the collection to that type, and all the built in collection types all have reference equality semantics (even `ImmutableArray`).

At the time I wasn't sure what the source generator's data model would look like, so I brought in those two helper classes. Given that the whole data model is just `ClassToGenerate`, I implemented `Equals()` and `GetHashCode()` myself and was able to remove both classes.